packages/identity/src/index.ts

```diff
@@ -99,6 +81,28 @@ export class Identity {
         return this._commitment
     }
 
+    /**
+     * Returns the private key encoded as a base64 string.
+     * @returns The private key as a base64 string.
+     */
+    public export(): string {
+        if (isString(this._privateKey)) {
+            return textToBase64(this._privateKey as string)
+        }
+
+        return bufferToBase64(this.privateKey as Buffer | Uint8Array)
+    }
+
+    /**
+     * Returns a Semaphore identity based on a private key encoded as a base64 string.
+     * The private key will be converted to a buffer, regardless of its original type.
```

Comment:
This is a bit surprising, though understandable since the Base64 doesn't preserve the original type.
There needs to be a unit test to ensure this case still round-trips to produce the same identity, not just now, but also in future, since the behavior which controls that is inside of the EdDSAPoseidon constructor.

Reply:
I think I personally would do 1, or nothing at all.  2 seems unnecessary if the type doesn't actually change behavior.  I.e. if buffer or string produce the same public key, then it's pretty safe to not propagate the specific TypeScript type of the object the user provided.

Reply:
Let's keep it as it is then 👍🏽